File: 0443-string-compression/0443-string-compression.cpp
class Solution {
public:
    int numOfDigits(int n) {
        if (n == 0) {
            return 1;
        }
        int res = 0;
        while (n > 0) {
            res++;
            n /= 10;
        }
        return res;
    }
    
    int compress(vector<char>& chars) {
        int newLen = 0;
        int currLen = size(chars);
        if (currLen == 1) {
            return 1;
        }
        int i = 0;
        int it = 0;
        while (i < currLen) {
            int count = 1;
            char c = chars[i];
            while (i < currLen - 1 && (chars[i] == chars[i + 1])) {
                count++;
                i++;
            }
            chars[it++] = c;
            int numDigits = numOfDigits(count);
            if (count != 1) {
                string strCount = to_string(count);
                for (char x : strCount) {
                    chars[it++] = x;
                }
                count += numDigits;
            }
            i++;
        }
        for (int i = it; i < currLen; i++) {
            chars.pop_back();
        }
        return it;
    }
};

File: 0443-string-compression/0443-string-compression.java
class Solution {
    public int compress(char[] chars) {
        int len = chars.length, iter = 0;
        for (int i = 0; i < len; i++) {
            char c = chars[i];
            int count = 0;
            while (i < len && chars[i] == c) {
                count++;
                i++;
            }
            if (i < len) {
                i--;
            }
            chars[iter++] = c;
            if (count > 1) {
                String strCount = String.valueOf(count);
                for (int k = 0; k < strCount.length(); k++) {
                    chars[iter++] = strCount.charAt(k);
                }
            }
        }
        return iter;
    }
}